### ISSUE TITLE
bug:  wrong event if UNTIL< DTSTART

### ISSUE THREAD

**Issue by zoifar** [niccokunzmann/python-recurring-ical-events#117]
**Describe the bug**
 
If UNTIL< DTSTART then a wrong event is returned.

**To Reproduce**
```
ical_string="""BEGIN:VCALENDAR
BEGIN:VEVENT
DTSTAMP:20231102T221721Z
DTSTART;VALUE=DATE:20231002
DTEND;VALUE=DATE:20231009
SUMMARY:test123
CATEGORIES:other
SUBCALENDAR-NAME:test
EVENT-ID:538924
EVENT-ALLDAY:true
RRULE:FREQ=WEEKLY;UNTIL=20231001;INTERVAL=2;BYDAY=MO
CREATED:20231102T221633Z
LAST-MODIFIED:20231102T221716Z
TRANSP:TRANSPARENT
STATUS:CONFIRMED
END:VEVENT
END:VCALENDAR"""
calendar = icalendar.Calendar.from_ical(ical_string)
events=recurring_ical_events.of(calendar).at(2023)

for event in events:
    start = event["DTSTART"].dt
    duration = event["DTEND"].dt - event["DTSTART"].dt
    print("start {} duration {}".format(start, duration))
```

 

**ICS file**
 

```
BEGIN:VCALENDAR
BEGIN:VEVENT
DTSTAMP:20231102T221721Z
DTSTART;VALUE=DATE:20231002
DTEND;VALUE=DATE:20231009
SUMMARY:test123
CATEGORIES:other
SUBCALENDAR-NAME:test
EVENT-ID:538924
EVENT-ALLDAY:true
RRULE:FREQ=WEEKLY;UNTIL=20231001;INTERVAL=2;BYDAY=MO
CREATED:20231102T221633Z
LAST-MODIFIED:20231102T221716Z
TRANSP:TRANSPARENT
STATUS:CONFIRMED
END:VEVENT
END:VCALENDAR
```

**Expected behavior**
No events should be returned.
 

**Console output**
`start 2023-10-02 duration 7 days, 0:00:00`
 

**Version:**
 
2.1.0

<small>
<hr/>
We're using <a href="https://polar.sh/niccokunzmann">Polar.sh</a> so you can upvote and help fund this issue. We receive the funding once the issue is completed & confirmed by you. Thank you in advance for helping prioritize & fund our work.
</small>

<a href="https://polar.sh/niccokunzmann/python-recurring-ical-events/issues/117">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/niccokunzmann/python-recurring-ical-events/issues/117/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/niccokunzmann/python-recurring-ical-events/issues/117/pledge.svg">
</picture>
</a>

**Comment by niccokunzmann** [comment]
@zoifar thanks for reporting this. It seems unusual to not remove the event at all. Can you tell me more about the background of where this is generated? The until value often is on a valid repetition date, in this example it is not. I just want to be sure that when this becomes expected behavior that there is a general expectation of ICS functioning like this. Maybe, there is something in the spec... If it is a single persons expectation, it can still be implemented.

**Comment by zoifar** [comment]
This is generated by atlassian confluence calendar if first event in series and all future instances are deleted.

**Comment by niccokunzmann** [comment]
Thanks, that makes sense to me. Would you like to give it a go?

**Comment by niccokunzmann** [comment]
What happens if you edit a recurrence and then delete the whole event series?

**Comment by zoifar** [comment]
> What happens if you edit a recurrence and then delete the whole event series?

If you edit a recurrence and then delete the whole event series, than no events are present in ICS file.

**Comment by niccokunzmann** [comment]
I am not sure if all special cases are covered as there might also be edited recurrences. Either way, you can test with
```
pip install git+https://github.com/niccokunzmann/python-recurring-ical-events
```
I think, this issue can be closed and a new version released.
What are your thoughts?

**Comment by zoifar** [comment]
It works. Thank you. 

I found one more case that returns wrong event:
```BEGIN:VCALENDAR
BEGIN:VEVENT
DTSTAMP:20231119T234504Z
DTSTART;VALUE=DATE:20231001
DTEND;VALUE=DATE:20231009
SUMMARY:test123
CATEGORIES:other
SUBCALENDAR-NAME:teeest
EVENT-ID:540467
EVENT-ALLDAY:true
RRULE:FREQ=WEEKLY;UNTIL=20231001;INTERVAL=1;BYDAY=MO
CREATED:20231119T234338Z
LAST-MODIFIED:20231119T234338Z
TRANSP:TRANSPARENT
STATUS:CONFIRMED
END:VEVENT
END:VCALENDAR
```
I suppose a separate bug should be submitted for that.

**Comment by niccokunzmann** [comment]
This is included in 2.1.1
